### PR TITLE
[Perf] Skip LTX-2.3 denoise masks when connectors use learned registers

### PR DESCRIPTION
**Pull request by mglyn**
# [Perf] Skip LTX-2.3 denoise masks when connectors use learned registers

## Purpose

Improve LTX-2.3 text-to-video compile performance by skipping the connector attention mask in the denoise transformer when the LTX-2.3 connectors replace padding tokens with learned registers.

This follows up on the graph-break investigation from #4790, but uses a smaller LTX-2.3-local implementation instead of changing `setup_compile()` or process-wide Dynamo settings.

LTX-2.3 connectors expose learned registers for both video and audio conditioning:

- `video_connector_num_learnable_registers`
- `audio_connector_num_learnable_registers`

With those registers enabled, padded prompt positions are converted into learned conditioning tokens by the connector. The denoise transformer no longer needs the original text padding mask for `attn2` / `audio_attn2`.

This is not just a compile workaround. In the learned-register connector path, padding positions are first replaced by learned register embeddings and the connector returns an all-valid downstream mask. Passing `None` to the denoise transformer is therefore equivalent to passing that all-valid mask, while avoiding the masked-attention bookkeeping.

Keeping the mask forces the FlashAttention masked varlen path inside the compiled LTX-2.3 transformer block. That path performs tensor-value-dependent mask classification and varlen metadata work, which prevents the repeated block from being captured as a cleaner compiled region and adds extra work in every denoise step.

This change keeps the optimization local to LTX-2.3:

- Detect whether both LTX-2.3 connectors use learned registers.
- Pass `None` as the denoise `encoder_attention_mask` /
  `audio_encoder_attention_mask` when learned registers are present.
- Preserve the old masked behavior when connectors do not expose learned
  registers.
- Keep the diffusion attention backends and shared attention metadata unchanged.

The safety boundary is intentionally narrow: if either connector does not expose
learned registers, the previous mask path is kept unchanged.

## Test Plan

Platform:

- 4xA100 80GB

Model:

- LTX-2.3 Diffusers

Workload:

```bash
export PROMPT="Floating crystal islands in cosmic starry sky, glowing nebula, soft luminous particles flowing around, slow camera rotation"
export NEG_PROMPT="low quality, blurry, noise, watermark, text, deformed figures, cartoon style, over-saturated color, frame jump"
```

Benchmark command:

```bash
cd /root/vllm-omni

# Baseline commit / branch without this change.
LABEL=before MODEL=/data/models/Lightricks/LTX-2.3-diffusers \
bash ./run_ltx23_mask_elision_sweep_a100.sh

# Patched commit / branch with this change.
LABEL=after MODEL=/data/models/Lightricks/LTX-2.3-diffusers \
bash ./run_ltx23_mask_elision_sweep_a100.sh
```

Benchmark settings:

- `exec=compile`
- `profile=base,cfg2`
- `cases=512x384x25f,1024x576x81f`
- `num_inference_steps=10`
- same-shape warmup before measured requests
- 10 measured requests per cell
- `--enable-diffusion-pipeline-profiler` disabled
- primary metric: `stage_0_gen_ms`

**vLLM Version:**

0.23.0

**vLLM-Omni Commit:**

fe04c217a4b53719844b2c624143f87861e5e172

## Test Result

Stage-0 generation time:

| exec | profile | case | before mean ms | after mean ms | delta ms | delta % | before peak MB | after peak MB | completed | failed |
|---|---|---|---:|---:|---:|---:|---:|---:|---:|---:|
| compile | base | 512x384_25f | 4017.23 | 3436.99 | -580.23 | -14.44% | 73258.0 | 73258.0 | 10 | 0 |
| compile | cfg2 | 512x384_25f | 3208.70 | 2419.56 | -789.14 | -24.59% | 73258.0 | 73258.0 | 10 | 0 |
| compile | base | 1024x576_81f | 21873.91 | 21488.61 | -385.30 | -1.76% | 76622.0 | 76619.4 | 10 | 0 |
| compile | cfg2 | 1024x576_81f | 12524.78 | 12087.01 | -437.77 | -3.50% | 76616.0 | 76616.0 | 10 | 0 |

Peak memory is effectively unchanged across the measured cells.

Trace validation:
---

8f0a6e27-e781-49bc-9674-be552efdb61c

Before: LTX-2.3 repeated transformer block enters the masked FlashAttention varlen path for prompt cross-attention.

---

{4A5FD8E8-A239-4B87-9C43-5E916BFC1676}

After: the denoise transformer receives no connector attention mask when learned registers are enabled, avoiding the masked varlen path for those prompt cross-attention calls.

---

**BEFORE SUBMITTING:** read [CONTRIBUTING.md](https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md) and run the [precheck-pr skill](https://github.com/vllm-project/vllm-omni/blob/main/.claude/skills/precheck-pr/SKILL.md) with the code agent for a self-check against project conventions.
(anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by mglyn** [comment]
This issue doesn't exist anymore after #5148 was merged.
